tests/test_browser.go

```diff
@@ -77,6 +80,16 @@ func newTestBrowser(tb testing.TB, opts ...any) *testBrowser {
 			skipClose = true
 		case withSamplesListener:
 			samples = opt
+		case withLookupFunc:
+			lookupFunc = func(key string) (string, bool) {
+				v, ok := opt(key)
+				if ok {
+					return v, ok
+				}
+				// return from the real environment lookup function
+				// so that we can debug (or other things) when we want it.
+				return os.LookupEnv(key)
```

Comment:
The behaviour could be quite confusing if an env var is set in the os (maybe for a test script test run) and then the unit/integration tests pick that up. If we don't need this now, then it might be better not to look up the os env.

Reply:
To be honest I find this a little bit confusing also. If I explicitly set the `withLookupFunc` option for `testbrowser` I would expect it to just use the function implementation that I passed, not default to O.S. env lookup if one variable is not set in my "mock" implementation. I think this can lead to confusing executions, but no strong opinion.

Reply:
If we remove this, it'll be harder to see logs and debug, at least for me 😄 I'm open to other suggestions 🙇 

In practice, we won't pass an environment variable other than `K6_BROWSER_LOG`. So, my other suggestion is to only accept `K6_BROWSER_LOG` here instead of accepting every environment variable. WDYT?